packages/components/src/autocomplete/index.js

```diff
@@ -471,12 +477,21 @@ function Autocomplete( {
 			return;
 		}
 
+		// If we don't have any matching filteredOptions from the last render iteration +
+		// we didn't have a new trigger typed, then we should not continue with this effect.
+		const mismatch = filteredOptions.length === 0;
+		if (mismatch && text.slice(-1) !== completer.triggerPrefix) {
+			return;
```

Comment:
Do we need to `reset()` prior to `return`ing? 🤔 

```suggestion
			reset();
			return;
```

Reply:
This is not relevant anymore as I changed and simplified the implementation a bit.